addons/mail_client_extension/controllers/main.py

```diff
@@ -182,59 +187,84 @@ def _create_company_from_iap(self, domain):
             values=iap_data,
             subtype_id=request.env.ref('mail.mt_note').id,
         )
-        
+
         return new_company, {'type': 'company_created'}
 
+    def _get_partner_dict(self, partner):
+        return {
+            'id': partner.id,
+            'name': partner.name,
+            'title': partner.function,
+            'email': partner.email,
+            'image': partner.image_128,
+            'phone': partner.phone,
+            'mobile': partner.mobile,
+            'enrichment_info': None,
+        }
+
     @http.route('/mail_client_extension/partner/get', type="json", auth="outlook", cors="*")
-    def res_partner_get_by_email(self, email, name, **kwargs):
+    def res_partner_get(self, email=None, name=None, partner_id=None, **kwargs):
+
+        assert partner_id or (name and email)
+
         response = {}
+        partner = None
+        if partner_id:
+            partner = request.env['res.partner'].browse(partner_id)
+            email = partner.email
 
-        #compute the sender's domain
         normalized_email = tools.email_normalize(email)
         if not normalized_email:
             response['error'] = 'Bad email.'
             return response
         sender_domain = normalized_email.split('@')[1]
 
-        # Search for the partner based on the email.
-        # If multiple are found, take the first one.
-        partner = request.env['res.partner'].search([('email', 'in', [normalized_email, email])], limit=1)
+        if not partner_id:
+
+            # Search for the partner based on the email.
+            # If multiple are found, take the first one.
+            partner = request.env['res.partner'].search([('email', 'in', [normalized_email, email])], limit=1)
+
         if partner:
-            response['partner'] = {
-                'id': partner.id,
-                'name': partner.name,
-                'title': partner.function,
-                'email': partner.email,
-                'image': partner.image_128,
-                'phone': partner.phone,
-                'mobile': partner.mobile,
-                'enrichment_info': None
-            }
-            # if there is already a company for this partner, just take it without enrichment.
-            if partner.parent_id:
-                response['partner']['company'] = self._get_company_dict(partner.parent_id)
+            response['partner'] = self._get_partner_dict(partner)
+            response['partner']['extra_info'] = self._get_partner_extra_info(partner.id)
+            if partner.company_type == 'company':
+                response['partner']['company'] = self._get_company_dict(partner)
```

Comment:
We usually name those "_prepare_XXX_values".
So "_prepare_company_values" in this case would be good.